[redtailedfox] IP

redtailedfox opened the pull request:
# Sori Chatbot
> "Competitive Programming is the Goat" 

Sori helps you with making a list of events to be stored. Its,
- Text Based
- Easy commands to remember
- Fast to use

All you need to do this
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/admin/ip-w4.html)
2. double-click it
3. add your tasks
4. let it manage tasks for you 👍 

And it is **FREE**\

Features:

- [x] Managing Tasks
- [x] Finding Tasks by name
- [ ] Finding tasks by start time
- [ ] Finding tasks by end time

This code written in java which looks like this:
```java
public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, world!");
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

redtailedfox committed:
"Level-0"

redtailedfox committed:
"Level-1"

redtailedfox committed:
"add,list"

redtailedfox committed:
mark as done

redtailedfox committed:
mark as donefinal

redtailedfox committed:
todoeventdeadline

redtailedfox committed:
testing&errors

redtailedfox committed:
delete

redtailedfox committed:
automated text ui testing

redtailedfox committed:
added enums

redtailedfox committed:
added save

redtailedfox committed:
txt file

redtailedfox committed:
date time patch

redtailedfox committed:
More OOP, refactored into packages

redtailedfox committed:
refactored into packages

redtailedfox committed:
Merge remote-tracking branch 'origin/add-gradle-support'

redtailedfox committed:
junit testing

redtailedfox committed:
add jar file

redtailedfox committed:
add javadocs

redtailedfox committed:
Merge remote-tracking branch 'origin/A-JavaDoc'

redtailedfox committed:
coding standard

redtailedfox committed:
coding standard

redtailedfox committed:
Merge remote-tracking branch 'origin/A-CodingStandard'

redtailedfox committed:
added find

redtailedfox committed:
added find

redtailedfox committed:
Merge remote-tracking branch 'origin/A-Level9'

zannloo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,206 @@
+     * @param s the string
+     * @return the Deadline Tasked added
+     */
+    public Task addDeadline(String s) {
```
I like how the method names are verbs and can be easily understood.

zannloo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,206 @@
+    }
+
+    /**
+     * Add a todo task based on the description of the string
```
Perhaps the first sentence of the method header comment could start in the another form (Returns ..., Sends ..., Adds ...) :)

zannloo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,206 @@
+    private void respond(String s) {
+        try {
+            switch (Parser.parseCommand(s)) {
+                case LIST:
```
Not sure if I like that there is an indentation for case clauses.

zannloo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,24 @@
+package duke;
+
+
+public class Parser {
```
I like how the names for classes are nouns.

zannloo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,113 @@
+        Task t;
+        switch (arr[0]) {
+            case "E":
+                LocalDateTime a = LocalDateTime.parse(arr[3].trim(),
```
Perhaps a more intuitive variable name could be used here (eg fromDate)?

zannloo reviewed (commented):
LGTM:) Overall, I like how well organised your project was and that the code was well documented with JavaDocs.

redtailedfox committed:
added GUI

redtailedfox committed:
Merge remote-tracking branch 'origin/A-Level10'

redtailedfox committed:
OOP added (redo)

redtailedfox committed:
Merge remote-tracking branch 'origin/A-MoreOOP'

redtailedfox committed:
Packages added (redo)

redtailedfox committed:
Merge remote-tracking branch 'origin/A-Packages'

SinhaVedant commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,197 @@
+
+
+    /**
+     * Mark the idx^th element in the list.
```
Perhaps the first sentence of the header should be of form Marks instead of Mark?

SinhaVedant commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,197 @@
+     * @param idx the index of the task(1-index)
+     * @return the Task deleted
+     */
+
```
Should there be a blank line between the documentation block and method?

SinhaVedant commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,113 @@
+
+    /**
+     * Init the storage at the current filePath.
+     * @param filePath where the saved file is
```
Perhaps you can introduce an empty line between the description and parameter section?

SinhaVedant commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
```
I like your ordering of import statements here.

SinhaVedant reviewed (commented):
Overall I found your code very easy to read. Just some coding standard related issues to fix.

LINCHENYU2030S commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,197 @@
+    private Ui ui;
+
+
+    private static String DIRECTORY = "./data";
+    private static String PATH = "./data/duke.txt";
```
I like how constants are all uppercased. Perpaps doing "private static final " will be better?
```suggestion
    private static final String DIRECTORY = "./data";
    private static final String PATH = "./data/duke.txt";
```

LINCHENYU2030S commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,33 @@
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
i like how the naming is done in camelCase

LINCHENYU2030S commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,69 @@
+    private Scanner sc = new Scanner(System.in);
+
+    public void hello() {
+        System.out.println("__________________________________________________________");
```
Since the horizontal line is used frequently in our code, perpaps it will be better to declare it as a constant?

LINCHENYU2030S commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,76 @@
+import java.util.stream.Collectors;
+
+public class TaskList {
+    private ArrayList<Task> l = new ArrayList<>();
```
perhaps it will be better to name this variable in a more meaningful way , for example, tasks

LINCHENYU2030S reviewed (commented):
LGTM. just some nits. Overall, i like how you organize your code and the naming is consistent

redtailedfox committed:
add assertions

redtailedfox committed:
Merge pull request #1 from redtailedfox/A-Assertions

A-Assertions

redtailedfox committed:
Add code quality

redtailedfox committed:
Merge pull request #2 from redtailedfox/A-CodeQuality

A-CodeQuality

redtailedfox committed:
Detect duplicates function added

redtailedfox committed:
Merge remote-tracking branch 'origin/C-DetectDuplicates'

redtailedfox committed:
Added more functionality for detecting duplicates

redtailedfox committed:
extension complete

redtailedfox committed:
Add dectectduplicate functionality

Previously duplicated tasks could be added to list.

Adding ability for tasklist to check if task was
added prior and not add duplicate tasks.

Doing so to prevent user from accidentally adding
the same task and clutter the list.

redtailedfox committed:
tag

redtailedfox committed:
tag

redtailedfox committed:
Add User Guide and Ui.png.

Previously, users had no idea how to use the app.

User Guide helps users understand this product.

Adding user guide and Ui allows users to familiarise
themselves with the product. It is done with visual
aids for greatest efficiency.

redtailedfox committed:
Clean Up

redtailedfox committed:
clean up 2

redtailedfox committed:
clean up3
